### PR TITLE
Add troubleshooting guides to shards allocation actions

### PR DESCRIPTION
**Pull request by andreidan**

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityActionGuideTests.java [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+    }
+
+    public void testIncreaseShardLimitClusterSettingInTier() {
+        assertThat(ACTION_INCREASE_SHARD_LIMIT_CLUSTER_SETTING_LOOKUP.get(DataTier.DATA_HOT).helpURL(),
+            is(INCREASE_CLUSTER_SHARD_LIMIT_ACTION_GUIDE));
+        assertThat(ACTION_INCREASE_SHARD_LIMIT_CLUSTER_SETTING.helpURL(), is(INCREASE_CLUSTER_SHARD_LIMIT_ACTION_GUIDE));
+    }
+
+    public void testMigrateDataRequiredToDataTiers() {
+        assertThat(ACTION_MIGRATE_TIERS_AWAY_FROM_REQUIRE_DATA.helpURL(), is(MIGRATE_TO_TIERS_ACTION_GUIDE));
```
Comment (jbaiera) [inline server/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardsAvailabilityActionGuideTests.java:78]:
```suggestion
        assertThat(ACTION_MIGRATE_TIERS_AWAY_FROM_REQUIRE_DATA.helpURL(), is(MIGRATE_TO_TIERS_ACTION_GUIDE));
        assertThat(ACTION_MIGRATE_TIERS_AWAY_FROM_INCLUDE_DATA.helpURL(), is(MIGRATE_TO_TIERS_ACTION_GUIDE));```

**Review by jbaiera — APPROVED** [review]
🚢

**Comment by andreidan** [comment]
@elasticmachine run elasticsearch-ci/part-2

**Comment by andreidan** [comment]
@elasticmachine update branch

**Pull request merged**